mantaray/views.py

```diff
@@ -542,6 +547,36 @@ def on_part(self, nick: str, reason: str | None) -> None:
             show_in_gui=self.server_view.should_show_join_leave_message(nick),
         )
 
+    def on_kick(self, channel: str, kicker: str, kicked_nick: str, reason: str | None) -> None:
+        self.userlist.remove_user(kicked_nick)
+        if reason is None:
+            reason = ''
+        if kicked_nick == self.server_view.core.nick:
+            kicked_tag = 'self-nick'
+        else:
+           kicked_tag = 'other-nick'
+        if kicker == self.server_view.core.nick:
+            kicker_tag = 'self-nick'
+        else:
+            kicker_tag = 'other-nick'
+        if kicked_nick == self.server_view.core.nick:
+            self.add_message(
+                "*",
+                (kicker, [kicker_tag]),
+                (' has kicked you from ', ['error']),
+                (f'#{self.view_name}', ['channel']),
+                ('. You can still join by typing ', ['error']),
+                (f'/join {self.view_name}', ['pinged']))
+        else:
+            self.add_message(
+                "*",
+                (kicker, [kicker_tag]),
+                (' has kicked ' , []),
+                (kicked_nick, [kicked_tag]),
```

Comment:
This can never be `'self-nick'`, because there's a separate message for that case:

```suggestion
                (kicked_nick, ['other-nick']),
```

Once this suggestion is applied, you don't need the `kicked_tag` variable at all.